cfme/middleware/__init__.py

```diff
@@ -53,3 +68,63 @@ def download(extension):
         download_btn("Download as {}".format(extensions_mapping[extension]))
     except:
         raise ValueError("Unknown extention. check the extentions_mapping")
+
+
+def add_deployment(filename, runtime_name=None, enable_deploy=True, cancel=False):
```

Comment:
If it is not causing mess, as I said, it's ok for me.
Server extends new "Container" mixin with "add_deploymet", Deployment extends new "Deployable" mixin with rest of methods.